lib/segment/src/vector_storage/multi_dense/simple_multi_dense_vector_storage.rs

```diff
@@ -271,18 +268,22 @@ impl<T: PrimitiveVectorElement> MultiVectorStorage<T> for SimpleMultiDenseVector
 
     /// None if key is out of bounds
     fn get_multi_opt(&self, key: PointOffsetType) -> Option<TypedMultiDenseVectorRef<T>> {
-        self.vectors_metadata
-            .get(key as usize)
-            .map(|metadata| TypedMultiDenseVectorRef {
-                flattened_vectors: self.vectors.get_many(metadata.start, metadata.size),
+        self.vectors_metadata.get(key as usize).map(|metadata| {
+            let flattened_vectors = self
+                .vectors
+                .get_many(metadata.start, metadata.inner_vectors_count)
+                .unwrap_or_else(|| panic!("Vectors does not contain data for {:?}", metadata));
```

Comment:
do we need this panic here? I guess we can return None here

Reply:
My reasoning here is that `vectors_metadata` and `vectors` are inconsistent if the key exists in `metadata` but NOT in `vectors`.
Therefore this is a broken invariant.

WDYT?

Reply:
No concerns